### PR TITLE
Makefile: Split run into new prebuild and run-only targets

### PR DESCRIPTION
**Pull request by jrtc27**
This provides a simple interface for build scripts that want to pre-run the webpack and ts-compile steps (e.g. when producing a container image) ahead of time rather, without requiring them to copy the current set of commands needed for each part and hope they don't change.

Fixes https://github.com/compiler-explorer/compiler-explorer/issues/3287

### CODE REVIEW

**Review by RubenRBS — APPROVED** [review]
This is a pretty nice solution, great idea! This looks ok to me but I'll let someone else take a look at it before merging

Thanks a lot! :)

**Review by jeremy-rifkin — APPROVED** [review]
Looks good to me

File: Makefile
Comment (mattgodbolt) [inline Makefile:83]:
```suggestion
run: prebuild   ## Runs the site like it runs in production
```

and then we only re-invoke make for `run-only` here?

Comment (jrtc27) [reply Makefile:83]: Don't think it makes much difference so can do if you'd rather, just felt it was a tiny bit cleaner to be consistent between the two stages

Comment (mattgodbolt) [reply Makefile:83]: Fair enough! Will merge as-is! thanks!

**Review by mattgodbolt — APPROVED** [review]

**Comment by jeremy-rifkin** [comment]
This is now live

**Pull request merged**
